### PR TITLE
feat: Token dropdown

### LINKED ISSUE #665: Add Token dropdown
TODO:
- Add per token rate to the redux state
- Bid per token (send token address to bid)
- Add the current selected token to the redux estate
- Select token via query string

### PR DESCRIPTION
**Pull request by cazala**
Closes #665

### CODE REVIEW

File: webapp/src/components/AuctionPage/AuctionPage.js [hunk 1]
```diff
@@ -136,7 +148,9 @@ export default class AuctionPage extends React.PureComponent {
 
   hasReachedLimit(selected) {
     const { landsLimitPerBid } = this.props.auctionParams
-    return Object.keys(selected).length >= landsLimitPerBid
+    const selectedAmount = Object.keys(selected).length
+    debugger
```
Comment (NicoSantangelo) [inline webapp/src/components/AuctionPage/AuctionPage.js]: Whoops

File: webapp/src/modules/auction/sagas.js [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
 import { api } from 'lib/api'
-import { getParams } from './selectors'
-
 import { splitCoodinatePairs } from 'shared/coordinates'
 
+import { getParams, getSelectedToken } from './selectors'
+import { TOKENS } from './utils'
+
 const ONE_BILLION = 1000000000 // 1.000.000.000
+const REFRESH_INTERVAL = 5000 // five segundos
```
Comment (NicoSantangelo) [inline webapp/src/modules/auction/sagas.js]: `segundos` intentional 🦄 ?

File: webapp/src/components/AuctionPage/AuctionPage.css [hunk 3]
```diff
@@ -1,8 +1,16 @@
+.AuctionPage {
+  padding-top: 0px !important;
```
Comment (NicoSantangelo) [inline webapp/src/components/AuctionPage/AuctionPage.css]: I was going to ask you if doing `.Page > .AuctionPage { padding: 0; }` (without an `!important`) was better, but checking the rest of the pages we also do

image
and
image

maybe it's a good idea to remove that padding and adding it when we need it, like in the /marketplace and profile page?

File: webapp/src/components/AuctionPage/Token/Token.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      return (
+        <Header size="large" className="Token">
+          <span className="loading">--.--</span> {symbol}
+        </Header>
+      )
+    }
+    return (
+      <Header size="large" className="Token">
+        {symbol === 'MANA' ? (
+          <span>⏣ {amount}</span>
```
Comment (NicoSantangelo) [inline webapp/src/components/AuctionPage/Token/Token.js]: This Mana looks awesome but isn't it confusing to have it just here? the rest of the site uses the old one

File: webapp/src/components/AuctionPage/TokenDropdown/TokenDropdown.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { SYMBOLS } from 'modules/auction/utils'
+
+import './TokenDropdown.css'
+
+const OPTIONS = SYMBOLS.map(symbol => ({
+  text: symbol,
+  value: symbol
+}))
+
+export default class TokenDropdown extends React.PureComponent {
```
Comment (NicoSantangelo) [inline webapp/src/components/AuctionPage/TokenDropdown/TokenDropdown.js]: _Not critical_, can we add PropTypes here too?

File: webapp/src/modules/auction/utils.js [hunk 6]
```diff
@@ -1,5 +1,14 @@
+import { env } from 'decentraland-commons'
 import { getLocalStorage } from '@dapps/lib/localStorage'
 
+export const TOKENS = {
+  MANA: env.get('REACT_APP_MANA_TOKEN_CONTRACT_ADDRESS'),
+  ZIL: env.get('REACT_APP_ZIL_TOKEN_CONTRACT_ADDRESS'),
+  DAI: env.get('REACT_APP_DAI_TOKEN_CONTRACT_ADDRESS')
+}
+
+export const SYMBOLS = Object.keys(TOKENS)
```
Comment (NicoSantangelo) [inline webapp/src/modules/auction/utils.js]: I think this variable might be a bit clearer as `TOKEN_SYMBOLS` when used from outside this file. The first time I saw `import { SYMBOL } from 'modules/auction/utils` I wasn't entirely sure what was the value.
That said it's kind of a big refactor so feel free to dismiss this comment

File: webapp/src/modules/auction/sagas.js [hunk 7]
```diff
@@ -71,3 +90,49 @@ function* handleBidRequest(action) {
...
+      )
+    }
+    yield put(fetchAuctionRateSuccess(selectedToken, rate))
+  } catch (error) {
+    yield put(fetchAuctionRateFailure(selectedToken, error.message))
+  }
+}
+
+function* handleFetchAuctionRateSuccess(action) {
+  const token = yield select(state => getSelectedToken(state))
```
Comment (NicoSantangelo) [inline webapp/src/modules/auction/sagas.js]: Just to be consistent
```suggestion
  const token = yield select(getSelectedToken)
```

File: webapp/src/modules/auction/sagas.js [hunk 8]
```diff
@@ -71,3 +90,49 @@ function* handleBidRequest(action) {
     yield put(bidOnParcelsFailure(error.message))
   }
 }
+
+function* handleFetchAuctionRateRequest(action) {
+  const { token } = action
+  const selectedToken = yield select(state => getSelectedToken(state))
```
Comment (NicoSantangelo) [inline webapp/src/modules/auction/sagas.js]: Just to be consistent
```suggestion
  const selectedToken = yield select(getSelectedToken)
```

File: webapp/src/modules/auction/sagas.js [hunk 9]
```diff
@@ -71,3 +90,49 @@ function* handleBidRequest(action) {
...
+function* handleConnectWalletSuccess(action) {
+  let connected = true
+  // keep refreshing params and rate while the user is on /auction
+  while (connected) {
+    const pathname = yield select(state => getPathname(state))
+    if (pathname === locations.auction()) {
+      yield put(fetchAuctionParamsRequest())
+    }
+    yield delay(REFRESH_INTERVAL)
+    connected = yield select(state => isConnected(state))
```
Comment (NicoSantangelo) [inline webapp/src/modules/auction/sagas.js]: Same as above with the `select` calls

This is really cool btw, I like checking the location from here

File: webapp/src/modules/auction/sagas.js (hunk 2 above)
Comment (cazala) [reply webapp/src/modules/auction/sagas.js]: cinco seconds

File: webapp/src/modules/auction/sagas.js (hunk 8 above)
Comment (cazala) [reply webapp/src/modules/auction/sagas.js]: ✅ done

File: webapp/src/modules/auction/sagas.js (hunk 7 above)
Comment (cazala) [reply webapp/src/modules/auction/sagas.js]: ✅ done

File: webapp/src/modules/auction/utils.js (hunk 6 above)
Comment (cazala) [reply webapp/src/modules/auction/utils.js]: ✅ done

File: webapp/src/components/AuctionPage/TokenDropdown/TokenDropdown.js (hunk 5 above)
Comment (cazala) [reply webapp/src/components/AuctionPage/TokenDropdown/TokenDropdown.js]: ✅ done

File: webapp/src/components/AuctionPage/AuctionPage.css (hunk 3 above)
Comment (cazala) [reply webapp/src/components/AuctionPage/AuctionPage.css]: mm i think from all the children of `<Page>` there's only 3 that don't need the padding, but the other +20 pages do:

screen shot 2018-11-22 at 10 46 52 am

File: webapp/src/components/ActivityPage/ActivityPage.container.js [hunk 10]
```diff
@@ -7,10 +7,11 @@ import {
   getAddress,
   isConnecting,
   isConnected
-} from 'modules/wallet/selectors'
-import { getAuthorizations } from 'modules/authorization/selectors'
+} from '@dapps/modules/wallet/selectors'
 import { clearTransactions } from '@dapps/modules/transaction/actions'
 
+import { getAuthorizations } from 'modules/authorization/selectors'
+
```
[the remark this answers is not in the document]
Comment (cazala) [reply webapp/src/components/ActivityPage/ActivityPage.container.js]: ✅ done

File: webapp/src/modules/auction/sagas.js (hunk 9 above)
Comment (cazala) [reply webapp/src/modules/auction/sagas.js]: ✅ done

File: webapp/src/components/AuctionPage/AuctionPage.js (hunk 1 above)
Comment (cazala) [reply webapp/src/components/AuctionPage/AuctionPage.js]: 😅

**Comment by cazala** [comment]
@NicoSantangelo thanks for the review! feedback addressed ✅

File: webapp/src/components/AuctionPage/Token/Token.js (hunk 4 above)
Comment (NicoSantangelo) [reply webapp/src/components/AuctionPage/Token/Token.js]: Gotcha

**Pull request merged**